### PR TITLE
Add new privileged functions for updater and merge ftp changes

### PR DESCRIPTION
**Pull request by Taylorr82**

### CODE REVIEW

File: main/main.c [hunk 1]
```diff
@@ -75,19 +77,18 @@ int ex2_main(int argc, char **argv) {
   InitIO();
 
   /* Initialization routine */
-  //init_filesystem();
+  init_filesystem();
```
Comment (ArashYzd) [inline main/main.c:80]: Would it run fine for people without an SD card? I think it'd be better commented out on the master branch.

File: main/main.c [hunk 2]
```diff
@@ -66,6 +66,8 @@ static void init_software();
 static inline SAT_returnState init_csp_interface();
 static void init_system_tasks();
 void vAssertCalled(unsigned long ulLine, const char *const pcFileName);
+static FTP app;
```
Comment (ArashYzd) [inline main/main.c:69]: What's this?

Comment (arrooney) [reply main/main.c:69]: this is Evan's FTP. Maybe we can give the variable a better name.

**Review by ArashYzd — APPROVED** [review]

File: source/HL_sys_link.cmd [hunk 3]
```diff
@@ -107,6 +107,9 @@ SECTIONS
 /* Misc                                                                       */
 
 /* USER CODE BEGIN (6) */
+#define GOLDEN_IMAGE
+//#define WORKING_IMAGE
+//#define BOOTLOADER_PRESENT
```
Comment (arrooney) [inline source/HL_sys_link.cmd:112]: Maybe we should explicitly undef these things?

**Review by arrooney — APPROVED** [review]

**Pull request merged**